src/envoy/http/service_control/client_cache.cc

```diff
@@ -387,6 +394,11 @@ void ClientCache::handleCheckResponse(const Status& http_status,
                 "Google Service Control Check is unavailable, and the "
                 "request is denied due to network fail closed, with error: {}",
                 final_status.error_message());
+
+      // If http_status is not ok, the Code::UNAVAILABLE is from http_status.
+      if (!http_status.ok()) {
```

Comment:
move this up to line 370? No need for duplicate if statement

Reply:
It is not duplicate. 

It needs satisfy both `http_status` is `UNAVALIABLE` and `network_fail_open_` is `false`. Moving to 370 needs checking the latter.

Reply:
Ack